models/userStatus.js

```diff
@@ -193,4 +205,133 @@ const updateAllUserStatus = async () => {
   }
 };
 
-module.exports = { deleteUserStatus, getUserStatus, getAllUserStatus, updateUserStatus, updateAllUserStatus };
+/**
+ * Updates the user status based on a new task assignment.
+ * @param {string} userId - The ID of the user.
+ * @returns {Promise<{
+ *   status: string,
+ *   message: string,
+ *   data?: object
+ * }>} - The response object indicating the status of the user status update.
+ * @throws {Error} If there is an error retrieving or updating the user status.
+ */
+const updateUserStatusOnNewTaskAssignment = async (userId) => {
+  try {
+    let latestStatusData;
+    try {
+      latestStatusData = await getUserStatus(userId);
+    } catch (error) {
+      logger.error("Unable to retrieve the current status" + error.message);
+      throw new Error("Unable to retrieve the current status");
+    }
+    const { userStatusExists } = latestStatusData;
+    if (!userStatusExists) {
+      return createUserStatusWithState(userId, userStatusModel, userState.ACTIVE);
+    }
+    const {
+      data: {
+        currentStatus: { state },
+      },
+    } = latestStatusData;
+    if (state === userState.ACTIVE) {
+      return generateAlreadyExistingStatusResponse(userState.ACTIVE);
+    }
+    if (state === userState.IDLE || state === userState.ONBOARDING) {
+      return updateCurrentStatusToState(userStatusModel, latestStatusData, userState.ACTIVE);
+    }
+    if (state === userState.OOO) {
+      return updateFutureStatusToState(userStatusModel, latestStatusData, userState.ACTIVE);
+    }
+    throw new Error("Please reach out to the administrator as your user status is not recognized as valid.");
+  } catch (error) {
+    logger.error(`Error while updating the status for ${userId} - ${error.message}`);
+    throw error;
+  }
+};
+
+/**
+ * Updates the user status based on a task update.
+ * @param {string} userName - The username associated with the user.
+ * @returns {Promise<{
+ *   status: number,
+ *   message: string,
+ *   error: string,
+ * } | {
+ *   status: string,
+ *   message: string,
+ *   data: object
+ * }>} - The response object indicating the status of the user status update or an error.
+ */
+
+const updateUserStatusOnTaskUpdate = async (userName) => {
+  try {
+    const userId = await getUserIdFromUserName(userName, usersCollection);
+    const userStatusUpdate = await updateUserStatusOnNewTaskAssignment(userId);
+    return userStatusUpdate;
+  } catch (error) {
+    if (error instanceof NotFound) {
+      return {
+        status: 404,
+        error: "Not Found",
+        message: error.message,
+      };
+    }
+    return {
+      status: 500,
+      error: "Internal Server Error",
+      message: error.message,
+    };
+  }
+};
+
+const updateStatusOnTaskCompletion = async (userId) => {
```

Comment:
here we are trying to update the user's status when a task is completed, but it is not handling the case where the user does not have any active tasks. If the user does not have any active tasks, the function will try to update the user's status to IDLE, but the updateCurrentStatusToState function will throw an error because the user's current status is not OOO

Reply:
So if you see here  we have a single task assigned to user SooJK37gzjIZfFNH0tlL
![image](https://github.com/Real-Dev-Squad/website-backend/assets/97341921/caf52f76-69d6-41c3-9951-89ccd1ddcb5e)

On marking this task as completed. 
![image](https://github.com/Real-Dev-Squad/website-backend/assets/97341921/a0fdae0a-b6a1-4edb-8fc3-d532b2a643d2)

Since there were no other task assigned the user SooJK37gzjIZfFNH0tlL their status got updated to IDLE
![image](https://github.com/Real-Dev-Squad/website-backend/assets/97341921/92faaaba-b16b-4bb4-b001-f9ed0d1f6938)